### ISSUE TITLE
Requesting convenience method: getting the image from an Image widget

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#10893]
Today, to get an image from an image widget, you have to jump through a few steps:

```
    Image image = new Image.network('https://i.stack.imgur.com/lkd0a.png');
    Completer<ui.Image> completer = new Completer<ui.Image>();
    image.image
      .resolve(new ImageConfiguration())
      .addListener((ImageInfo info, bool _) => completer.complete(info.image));
```

It would be pretty nice if we had a convenience method like this:

```
    Image image = new Image.network('https://i.stack.imgur.com/lkd0a.png');
    ui.Image actualImage = await image.uiImage;
```

(strawman names, of course)

**Comment by Hixie** [comment]
Generally we don't consider widgets to be things you read from.

The expected code today would be:
```dart
ImageProvider image = new NetworkImage('https://i.stack.imgur.com/lkd0a.png');
Completer<ui.Image> completer = new Completer<ui.Image>();
image
  .resolve(new ImageConfiguration())
  .addListener((ImageInfo info, bool _) => completer.complete(info.image));
```

However, that wouldn't necessarily give you the right thing (nor would the code above). The actual image obtained could depend on e.g. the actual device pixel ratio (well, for network images we currently ignore the image configuration, but for asset images we don't).

What are you actually trying to do? Get the size of an image fetched from the network?

**Comment by sethladd** [comment]
> Get the size of an image fetched from the network?

Yes, that was the original question. Which I admit, is just a curiosity. If the answer is "don't use a widget if you want to get easy access to the ui.Image, instead here's the code you'll need: ..." then that's fine. We can document that.

**Comment by Hixie** [comment]
Ok we can add some documentation to the widget to say how to get the image without using the widget.

**Comment by pedromassangocode** [comment]
The actual documentation of `Image.network` doesn't says how to download an image.

https://api.flutter.dev/flutter/widgets/Image/Image.network.html

**Comment by cv692001** [comment]
Hey @sethladd , can you please help me to redirect where dartdocs are present . Actually I am new to programming 👍

**Comment by richard457** [comment]
Dart website can help you a lot here dart.dev, dartpad.dev and you might
check flutter too!

**Comment by danagbemava-nc** [comment]
The documentation for `Image.network` hasn't been updated with this information yet.

**Comment by Hixie** [comment]
This is blocked on https://github.com/flutter/flutter/issues/130524
